generators/app/templates/src/main/java/package/service/kafka/deserializer/StringDeserializer.java.ejs

```diff
@@ -0,0 +1,42 @@
+<%#
+ Copyright 2013-2020 the original author or authors from the JHipster project.
+
+ This file is part of the JHipster project, see https://www.jhipster.tech/
+ for more information.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+-%>
+package <%= packageName %>.service.kafka.deserializer;
+
+import io.vavr.control.Either;
+import org.apache.kafka.common.errors.SerializationException;
+import org.apache.kafka.common.serialization.Deserializer;
+import org.apache.kafka.common.serialization.Serdes;
+
+public class <%= entityClass %>Deserializer implements Deserializer<Either<DeserializationError, String>> {
```

Comment:
I thin we should only have `EntityDeserializer.java.ejs` and use `type === 'String'` or anything else to detect you're using no entity instead of having a `StringDeserializer.java.ejs` here

Reply:
EntityDeserializer and StringDeserializer don't have business code in common. By keeping them separated, these files behave like real templates (variables replacement). There is no 'if', 'else', ... that makes the template hard to read when it grows.
This way we also have the same segregation than Kafka (De)serializer such as IntegerDeserializer, UUIDDeserializer, DoubleDeserializer, etc...